packages/payment-detection/src/erc20/fee-proxy-contract.ts

```diff
@@ -18,16 +18,43 @@ class NetworkNotSupported extends Error {}
 /** Exception when version not supported */
 class VersionNotSupported extends Error {}
 
+export type deploymentInformationGetter = (
+  networkName: string,
+  artifactsVersion?: string,
+) => {
+  address: string;
+  creationBlockNumber: number;
+};
+
 /**
  * Handle payment networks with ERC20 fee proxy contract extension
  */
-export default class PaymentNetworkERC20FeeProxyContract implements PaymentTypes.IPaymentNetwork {
-  private extension: ExtensionTypes.PnFeeReferenceBased.IFeeReferenceBased;
+export default class PaymentNetworkERC20FeeProxyContract<
+  ExtensionType extends ExtensionTypes.PnFeeReferenceBased.IFeeReferenceBased = ExtensionTypes.PnFeeReferenceBased.IFeeReferenceBased
+> implements PaymentTypes.IPaymentNetwork {
+  protected extension: ExtensionType;
+  protected getDeploymentInformation: deploymentInformationGetter;
+  protected paymentNetworkId: ExtensionTypes.ID;
+
   /**
    * @param extension The advanced logic payment network extensions
    */
-  public constructor({ advancedLogic }: { advancedLogic: AdvancedLogicTypes.IAdvancedLogic }) {
-    this.extension = advancedLogic.extensions.feeProxyContractErc20;
+  public constructor({
+    advancedLogic,
+    extension,
+    getDeploymentInformation,
```

Comment:
this could be a protected method instead of being injected? it feels 100% linked to the payment network